platform/bukkit/plugin/src/main/java/me/lokka30/treasury/plugin/bukkit/hooks/Baltop.java

```diff
@@ -21,14 +23,16 @@
 import org.bukkit.OfflinePlayer;
 import org.bukkit.scheduler.BukkitRunnable;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 public class Baltop extends BukkitRunnable {
 
     private final boolean enabled;
-    private final int topSize, taskDelay;
+    private final int topSize;
+    private final int taskDelay;
     private final EconomyProvider provider;
 
-    private Multimap<String, TopPlayer> baltop;
+    private final Multimap<String, TopPlayer> baltop;
```

Comment:
wouldn't making it final require a lock?

Reply:
No, it's only initialized once. A final value is still mutable, it is just not reassignable. It already was not reassignable in the existing design, this was just a clarification of that.

Reply:
no the thing is that i know final maps require either a lock to get accessed or a `synchronized` block

Reply:
The `final` keyword exclusively prevents reassignment of a reference. It does nothing more and nothing less. It has nothing to do with concurrency. While the maps and collections used should likely be concurrent due to the fact that they're populated on a different thread, the keyword being present does not affect that.